src/Swarm/Game/Step.hs

```diff
@@ -110,7 +110,7 @@ gameTick = do
 
             when (newLoc /= oldLoc) $ do
               robotsByLocation . at oldLoc %= deleteOne rn
-              robotsByLocation . at newLoc . non Empty %= IS.insert rn
+              robotsByLocation . at newLoc . non Empty %= IS.insert rn -- TODO: should this be above?
```

Comment:
I'm not sure what you mean by "should be above", do you mean the `non Empty` should be on the previous line instead, with `oldLoc`?  I don't think that makes sense.  `at` yields a `Maybe Set`; `deleteOne` explicitly takes a `Maybe Set` because we want it to explicitly remove the mapping entirely if the set becomes empty.  On the other hand, the `non Empty` on the second line turns the `Maybe Set` into a `Set`, substituting an empty set for `Nothing`, so that if no mapping existed for the new location previously, it creates an empty set for us to insert `rn` into.

Reply:
Oh, I thought it would remove the empty set, so I had it backwards. Thanks for the explanation :slightly_smiling_face: